src/devtools/client/webconsole/actions/input.js

```diff
@@ -79,6 +75,8 @@ function evaluateExpression(expression) {
       return null;
     }
 
+    appendToStoredHistory(expression);
```

Comment:
okay - i think there's a nicer way to do this. We have a pattern in atleast the debugger, maybe the console already where we subscribe to store events and then update prefs there

https://github.com/RecordReplay/devtools/blob/master/src/ui/setup/prefs.ts#L29-L53

Reply:
Thanks. All for moving towards `sagas` or `XState`.

I'm not currently feeling the pain of the subscribe callback. I haven't seen the perf impact when measured and find it pretty simple in this one use case. Other side-effects (API calls, other thunks would be very bad though).